### ISSUE TITLE
Wrong IncorrectLinesCountAfterControlStructure

### ISSUE THREAD

**Issue by temp** [slevomat/coding-standard#1127]
I'm having a weird error, here is a testcase, derived from a real test-class in my code. The code makes no sense, I removed most of the lines to just show the problem.

It reports this error:
```
 21 | ERROR | [x] Expected 1 lines after "return", found 0.
    |       |     (SlevomatCodingStandard.ControlStructures.JumpStatementsSpacing.IncorrectLinesCountAfterControlStructure)
```

The error disappears when I add another function below, that returns something...
Also disappears when the lines before the return-statement are deleted.

```php
<?php

declare(strict_types=1);

namespace AppTests\Test;

use Lcobucci\JWT\Configuration;

use function assert;

/**
 * Test case.
 */
abstract class Test
{
    final protected function createToken(): callable
    {
        $configuration = self::$container->get(Configuration::class);
        assert($configuration instanceof Configuration);

        return static fn () => $configuration;
    }
}
```

**Comment by kukulich** [comment]
@temp What's your configuration for this sniff?

**Comment by temp** [comment]
@kukulich sorry, seem like I didn't get a reply-email from github :(

Our Ruleset is based on the doctrine coding-standard:

```xml
    <rule ref="SlevomatCodingStandard.ControlStructures.JumpStatementsSpacing">
        <properties>
            <property name="linesCountBeforeWhenFirstInCaseOrDefault" value="0"/>
            <property name="linesCountAfterWhenLastInCaseOrDefault" value="1"/>
            <property name="linesCountAfterWhenLastInLastCaseOrDefault" value="0"/>
            <property name="tokensToCheck" type="array">
                <element value="T_RETURN" />
                <element value="T_THROW" />
                <element value="T_YIELD" />
                <element value="T_YIELD_FROM" />
            </property>
        </properties>
    </rule>
```

**Comment by kukulich** [comment]
I'm not able to reproduce it: https://github.com/slevomat/coding-standard/commit/3724af6865fb2a3a383e04a140119ce38d5c39a8

So it's probably fixed in master.

**Comment by temp** [comment]
@kukulich sorry, but it doesn't seem to be fixed. This is somehow dependent from other code surrounding it. When I remove all other lines from jumpStatementsSpacingWithDefaultSettingsNoErrors.php, the test breaks.

```php
<?php // lint >= 7.4

abstract class Test
{
        final protected function createToken(): callable
        {
                $configuration = self::$container->get(Configuration::class);
                assert($configuration instanceof Configuration);

                return static fn () => $configuration;
        }
}
```

```
± vendor/phpunit/phpunit/phpunit
PHPUnit 9.5.0 by Sebastian Bergmann and contributors.

Warning:       No code coverage driver available
Warning:       Your XML configuration validates against a deprecated schema.
Suggestion:    Migrate your XML configuration using "--migrate-configuration"!

F

Time: 00:00.063, Memory: 24.00 MB

There was 1 failure:

1) SlevomatCodingStandard\Sniffs\ControlStructures\JumpStatementsSpacingSniffTest::testDefaultSettingsNoErrors
No errors expected, but 1 errors found.
Failed asserting that an array is empty.

/Users/swentz/git/coding-standard/SlevomatCodingStandard/Sniffs/TestCase.php:75
/Users/swentz/git/coding-standard/tests/Sniffs/ControlStructures/JumpStatementsSpacingSniffTest.php:13

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.
```

**Comment by kukulich** [comment]
Fixed in https://github.com/slevomat/coding-standard/commit/a4f7a07bc2a50d0c2a4f5664d98831eaace2478a
